open-craft/learner-profile-dashboard
# Add support for multiple LPDs

Cf. [SE-901](https://tasks.opencraft.com/browse/SE-901)

This PR adds support for defining and using multiple LPDs with the same deployment of the learner-profile-dashboard code.

This will allow course authors to keep learner profile data separate across different courses and/or course runs. It also:

* Avoids efforts to clear out and back up existing data from previous course runs.
* Preserves ability to access data from previous course runs via LPD database, Django shell, Django admin. No need to fall back on backups for this purpose.

**Test instructions**

Changes have been deployed to [HGSE stage](https://hgse-stage.opencraft.hosting/) for verification.

1. Create an account on HGSE stage and enroll in the demo course.

1. Navigate to [this unit](https://hgse-stage.opencraft.hosting/courses/course-v1:edX+DemoX+Demo_Course/jump_to/block-v1:edX+DemoX+Demo_Course+type@vertical+block@45c7cedb4bfe46f4a68c78787151cfb5) and make sure it loads without error.

1. Answer a few questions from the first section of the LPD and click "Submit your answers".

1. In another browser tab, navigate to [this unit](https://hgse-stage.opencraft.hosting/courses/course-v1:edX+DemoX+Demo_Course/jump_to/block-v1:edX+DemoX+Demo_Course+type@vertical+block@85d2d9021b1d461581e036016b68c7d0) and make sure it loads without error.

1. Answer the questions and click "Submit your answers".

1. Go back to the first tab and reload it. Confirm that the LPD displays the answers you provided earlier.

1. Go back to the second tab and reload it.  Confirm that the LPD displays the answers you provided earlier.

**Reviewers**
